src/components/AgentTree/index.tsx

```diff
@@ -256,6 +256,9 @@ class AgentTree extends React.Component<AgentTreeProps> {
             <div className={styles.container}>
                 <TreeNode headerContent={this.renderCheckAllButton()} />
                 {treeData.map((nodeData) => {
+                    const childrenHaveDifferentColors = nodeData.children.some(
+                        (child, _, arr) => child.color !== arr[0].color
+                    );
```

Comment:
for future reference: you should always convert to lowercase if you're comparing strings, colors often come in both casings